### PR TITLE
[Workspace] Fix dependency resolution in edit mode

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
- <rdar://problem/29944464> Package manager fails to do dependency
  resolution if some package is in edit mode

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
Depends on https://github.com/apple/swift-package-manager/pull/900

File: Sources/Workspace/Workspace.swift [hunk 1]
```diff
@@ -818,8 +843,10 @@ public class Workspace {
             let specifier = RepositorySpecifier(url: externalManifest.url)
```
Comment (ddunbar) [inline Sources/Workspace/Workspace.swift]: I think the FIXME above has now been addressed, too, FWIW?

File: Sources/Workspace/Workspace.swift [hunk 2]
```diff
@@ -874,10 +901,19 @@ public class Workspace {
...
+        guard var dependency = dependencyMap[specifier] else {
             fatalError("This should never happen, trying to remove \(specifier) which isn't in workspace")
         }
 
+        // If this dependency is based on a dependency, switch to that because we don't want to touch the editable checkout here.
+        //
+        // FIXME: This will remove also the data about the editable dependency and it will not be possible to "unedit" that dependency anymore.
+        // To do that we need to persist the value of isInEditableState and also store the package names in managed dependencies, because 
+        // it will not be possible to lookup these dependencies using their manifests as we won't have them anymore.
+        if let basedOn = dependency.basedOn {
```
Comment (ddunbar) [inline Sources/Workspace/Workspace.swift]: Maybe we should have a bug tracking this FIXME?

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Pull request merged**